File: LogicOverSyntaxCodes/Searching/PythonCodes/Leetcode1539.py
class Solution(object):
    def findKthPositive(self, arr, k):
        
        start, end = 0, len(arr) - 1

        while start <= end:
            mid = start + (end - start) // 2
            one_based_indexed = mid + 1
            missing_elements = arr[mid] - one_based_indexed
            
            if k > missing_elements:
                start = mid + 1
            else:
                end = mid - 1

        return start + k

File: LogicOverSyntaxCodes/Searching/JavaCodes/Leetcode1539.java
class Solution {
   public static int findKthPositive(int[] arr, int k) {
        int start=0,end=arr.length-1;
        
        while(start<=end){
            int mid=start+(end-start)/2;
            int oneBasedIndexed=mid+1;
            int missingElements=arr[mid]-oneBasedIndexed;
            if(k>missingElements){
                start=mid+1;
            }
            else{
                end=mid-1;
            }
        }
        return start+k;
    }
}